src/main/kotlin/inc/evil/d17map/services/ReservationService.kt

```diff
@@ -54,4 +54,22 @@ class ReservationService(
         val futureReservations = user?.reservations?.filter { it.date.isAfter(LocalDate.now()) }
         return futureReservations?.map { toReservationDto(it) }
     }
+
+    fun getReservationsForWeek(monday: LocalDate): List<ReservationDto> {
+        val endOfWeek = monday.plusDays(6)
+        val reservations = reservationRepository.findAllByDateBetween(monday, endOfWeek)
+        return reservations.map { toReservationDto(it) }
+    }
+
+    fun getUserWeekReservations(): List<ReservationDto> {
+        val loggedUserEmail = SecurityContextHolder.getContext().authentication?.name
+        val user = userRepository.findByEmail(loggedUserEmail ?: return emptyList())
+
+        val today = LocalDate.now()
+        val monday = today.with(java.time.temporal.TemporalAdjusters.previousOrSame(java.time.DayOfWeek.MONDAY))
```

Comment:
Wyjaśnisz co robi ta linijka?

Reply:
Ustawia "monday" na datę najbliższego poniedziałku w stosunku do dzisiejszej daty (jeśli dzisiaj mamy poniedziałek, to też to uwzględni)

Reply:
Nie da się jakoś ładniej zaimportować tej metody `previousOrSame`? i tak samo `DayOfWeek.MONDAY`?

Reply:
Faktycznie, poprawię - dodam importy